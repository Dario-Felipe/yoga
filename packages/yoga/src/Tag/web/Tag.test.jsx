import React from 'react';
import { render } from '@testing-library/react';
import { Building } from '@gympass/yoga-icons';

import { ThemeProvider, Tag, v3theme } from '../..';

describe('<Tag />', () => {
  it('should match snapshot when v3Theme is settled', () => {
    const { container } = render(
      <ThemeProvider theme={v3theme}>
        <Tag>v3Theme</Tag>
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot', () => {
    const { container } = render(
      <ThemeProvider>
        <Tag>default</Tag>
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with custom icon and informative type', () => {
    const { container } = render(
      <ThemeProvider>
        <Tag.Informative variant="success" icon={Building}>
          success with custom icon
        </Tag.Informative>
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with without icon and informative type', () => {
    const { container } = render(
      <ThemeProvider>
        <Tag.Informative variant="success">
          success without icon
        </Tag.Informative>
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with variant prop', () => {
    const { container } = render(
      <ThemeProvider>
        <Tag variant="success">success</Tag>
        <Tag variant="informative">informative</Tag>
        <Tag variant="attention">attention</Tag>
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with variant prop and informative type', () => {
    const { container } = render(
      <ThemeProvider>
        <Tag.Informative variant="success">success</Tag.Informative>
        <Tag.Informative variant="informative">informative</Tag.Informative>
        <Tag.Informative variant="attention">attention</Tag.Informative>
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with variant and margin prop ', () => {
    const { container } = render(
      <ThemeProvider>
        <Tag variant="success">success</Tag>
        <Tag mv="medium" variant="informative">
          informative
        </Tag>
        <Tag variant="attention">attention</Tag>
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with variant prop and informative type with margin prop', () => {
    const { container } = render(
      <ThemeProvider>
        <Tag.Informative variant="success">success</Tag.Informative>
        <Tag.Informative mv="medium" variant="informative">
          informative
        </Tag.Informative>
        <Tag.Informative variant="attention">attention</Tag.Informative>
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with ellipsis prop', () => {
    const { container } = render(
      <ThemeProvider>
        <Tag maxWidth="10px" ellipsis>
          Default with ellipsis
        </Tag>
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with variant, informative type and ellipsis prop', () => {
    const { container } = render(
      <ThemeProvider>
        <Tag.Informative variant="success" maxWidth="10px" ellipsis>
          Success with ellipsis
        </Tag.Informative>
        <Tag.Informative variant="attention" maxWidth="10px" ellipsis>
          Attention with ellipsis
        </Tag.Informative>
        <Tag.Informative variant="informative" maxWidth="10px" ellipsis>
          Informative with ellipsis
        </Tag.Informative>
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });
});
