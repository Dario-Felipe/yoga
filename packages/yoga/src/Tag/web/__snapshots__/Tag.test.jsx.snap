// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag /> should match snapshot 1`] = `
.c0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 8px 8px 8px 8px;
  background-color: rgba(107,107,120,0.25);
  color: #6B6B78;
  border-radius: 8px;
  font-size: 12px;
  font-weight: 700;
}

.c0 svg {
  margin-right: 5px;
}

<div>
  <div
    class="c0"
    color="#6B6B78"
  >
    <svg
      fill="#6B6B78"
      height="12"
      width="14"
    />
    default
  </div>
</div>
`;

exports[`<Tag /> should match snapshot with all variants 1`] = `
.c9 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 8px 8px 8px 8px;
  background-color: rgba(107,107,120,0.25);
  color: #6B6B78;
  border-radius: 8px;
  font-size: 12px;
  font-weight: 700;
}

.c9 svg {
  margin-right: 5px;
}

.c0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 8px 8px 8px 8px;
  background-color: rgba(35,27,34,0.25);
  color: #231B22;
  border-radius: 8px;
  font-size: 12px;
  font-weight: 700;
}

.c0 svg {
  margin-right: 5px;
}

.c1 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 8px 8px 8px 8px;
  background-color: rgba(216,56,94,0.25);
  color: #D8385E;
  border-radius: 8px;
  font-size: 12px;
  font-weight: 700;
}

.c1 svg {
  margin-right: 5px;
}

.c2 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 8px 8px 8px 8px;
  background-color: rgba(29,133,108,0.25);
  color: #1D856C;
  border-radius: 8px;
  font-size: 12px;
  font-weight: 700;
}

.c2 svg {
  margin-right: 5px;
}

.c3 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 8px 8px 8px 8px;
  background-color: rgba(255,194,76,0.25);
  color: #FFC24C;
  border-radius: 8px;
  font-size: 12px;
  font-weight: 700;
}

.c3 svg {
  margin-right: 5px;
}

.c4 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 8px 8px 8px 8px;
  background-color: rgba(112,104,212,0.25);
  color: #7068D4;
  border-radius: 8px;
  font-size: 12px;
  font-weight: 700;
}

.c4 svg {
  margin-right: 5px;
}

.c5 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 8px 8px 8px 8px;
  background-color: rgba(255,176,167,0.25);
  color: #FFB0A7;
  border-radius: 8px;
  font-size: 12px;
  font-weight: 700;
}

.c5 svg {
  margin-right: 5px;
}

.c6 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 8px 8px 8px 8px;
  background-color: rgba(255,135,76,0.25);
  color: #FF874C;
  border-radius: 8px;
  font-size: 12px;
  font-weight: 700;
}

.c6 svg {
  margin-right: 5px;
}

.c7 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 8px 8px 8px 8px;
  background-color: rgba(255,68,56,0.25);
  color: #FF4438;
  border-radius: 8px;
  font-size: 12px;
  font-weight: 700;
}

.c7 svg {
  margin-right: 5px;
}

.c8 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 8px 8px 8px 8px;
  background-color: rgba(113,2,82,0.25);
  color: #710252;
  border-radius: 8px;
  font-size: 12px;
  font-weight: 700;
}

.c8 svg {
  margin-right: 5px;
}

.c10 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 8px 8px 8px 8px;
  background-color: rgba(152,152,166,0.25);
  color: #9898A6;
  border-radius: 8px;
  font-size: 12px;
  font-weight: 700;
}

.c10 svg {
  margin-right: 5px;
}

.c11 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 8px 8px 8px 8px;
  background-color: rgba(215,215,224,0.25);
  color: #D7D7E0;
  border-radius: 8px;
  font-size: 12px;
  font-weight: 700;
}

.c11 svg {
  margin-right: 5px;
}

.c12 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 8px 8px 8px 8px;
  background-color: rgba(245,245,250,0.25);
  color: #F5F5FA;
  border-radius: 8px;
  font-size: 12px;
  font-weight: 700;
}

.c12 svg {
  margin-right: 5px;
}

.c13 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 8px 8px 8px 8px;
  background-color: rgba(255,255,255,0.25);
  color: #FFFFFF;
  border-radius: 8px;
  font-size: 12px;
  font-weight: 700;
}

.c13 svg {
  margin-right: 5px;
}

<div>
  <div
    class="c0"
    color="#231B22"
  >
    <svg
      fill="#231B22"
      height="12"
      width="14"
    />
    primary
  </div>
  <div
    class="c1"
    color="#D8385E"
  >
    <svg
      fill="#D8385E"
      height="12"
      width="14"
    />
    secondary
  </div>
  <div
    class="c1"
    color="#D8385E"
  >
    <svg
      fill="#D8385E"
      height="12"
      width="14"
    />
    vibin
  </div>
  <div
    class="c2"
    color="#1D856C"
  >
    <svg
      fill="#1D856C"
      height="12"
      width="14"
    />
    hope
  </div>
  <div
    class="c3"
    color="#FFC24C"
  >
    <svg
      fill="#FFC24C"
      height="12"
      width="14"
    />
    energy
  </div>
  <div
    class="c4"
    color="#7068D4"
  >
    <svg
      fill="#7068D4"
      height="12"
      width="14"
    />
    relax
  </div>
  <div
    class="c5"
    color="#FFB0A7"
  >
    <svg
      fill="#FFB0A7"
      height="12"
      width="14"
    />
    peace
  </div>
  <div
    class="c6"
    color="#FF874C"
  >
    <svg
      fill="#FF874C"
      height="12"
      width="14"
    />
    verve
  </div>
  <div
    class="c7"
    color="#FF4438"
  >
    <svg
      fill="#FF4438"
      height="12"
      width="14"
    />
    uplift
  </div>
  <div
    class="c8"
    color="#710252"
  >
    <svg
      fill="#710252"
      height="12"
      width="14"
    />
    deepPurple
  </div>
  <div
    class="c0"
    color="#231B22"
  >
    <svg
      fill="#231B22"
      height="12"
      width="14"
    />
    stamina
  </div>
  <div
    class="c9"
    color="#6B6B78"
  >
    <svg
      fill="#6B6B78"
      height="12"
      width="14"
    />
    dark
  </div>
  <div
    class="c10"
    color="#9898A6"
  >
    <svg
      fill="#9898A6"
      height="12"
      width="14"
    />
    medium
  </div>
  <div
    class="c9"
    color="#6B6B78"
  >
    <svg
      fill="#6B6B78"
      height="12"
      width="14"
    />
    deep
  </div>
  <div
    class="c11"
    color="#D7D7E0"
  >
    <svg
      fill="#D7D7E0"
      height="12"
      width="14"
    />
    light
  </div>
  <div
    class="c12"
    color="#F5F5FA"
  >
    <svg
      fill="#F5F5FA"
      height="12"
      width="14"
    />
    clear
  </div>
  <div
    class="c13"
    color="#FFFFFF"
  >
    <svg
      fill="#FFFFFF"
      height="12"
      width="14"
    />
    white
  </div>
</div>
`;

exports[`<Tag /> should match snapshot with custom icon 1`] = `
.c0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 8px 8px 8px 8px;
  background-color: rgba(107,107,120,0.25);
  color: #6B6B78;
  border-radius: 8px;
  font-size: 12px;
  font-weight: 700;
}

.c0 svg {
  margin-right: 5px;
}

<div>
  <div
    class="c0"
    color="#6B6B78"
  >
    <svg
      fill="#6B6B78"
      height="12"
      width="14"
    />
    default with custom icon
  </div>
</div>
`;

exports[`<Tag /> should match snapshot with full prop 1`] = `
.c0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 8px 8px 8px 8px;
  background-color: rgba(107,107,120,0.25);
  color: #6B6B78;
  border-radius: 8px;
  font-size: 12px;
  font-weight: 700;
}

.c0 svg {
  margin-right: 5px;
}

<div>
  <div
    class="c0"
    color="#6B6B78"
  >
    <svg
      fill="#6B6B78"
      height="12"
      width="14"
    />
    full
  </div>
</div>
`;

exports[`<Tag /> should match snapshot with without icon 1`] = `
.c0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 8px 8px 8px 8px;
  background-color: rgba(107,107,120,0.25);
  color: #6B6B78;
  border-radius: 8px;
  font-size: 12px;
  font-weight: 700;
}

.c0 svg {
  margin-right: 5px;
}

<div>
  <div
    class="c0"
    color="#6B6B78"
  >
    default without icon
  </div>
</div>
`;
